org.eclipse.gef.examples.text/src/org/eclipse/gef/examples/text/SelectionRange.java

```diff
@@ -86,78 +86,75 @@ public SelectionRange(TextEditPart begin, int bo, TextEditPart end, int eo) {
 		this(new TextLocation(begin, bo), new TextLocation(end, eo));
 	}
 
-	private void depthFirstTraversal(EditPart part, ArrayList result) {
+	private static void depthFirstTraversal(EditPart part, List<EditPart> result) {
 		if (part.getChildren().isEmpty())
 			result.add(part);
 		else
 			for (int i = 0; i < part.getChildren().size(); i++)
-				depthFirstTraversal((EditPart) part.getChildren().get(i), result);
+				depthFirstTraversal(part.getChildren().get(i), result);
 	}
 
+	@Override
 	public boolean equals(Object obj) {
-		if (obj instanceof SelectionRange) {
-			SelectionRange other = (SelectionRange) obj;
+		if (obj instanceof SelectionRange other) {
 			return other.begin == begin && other.end == end && other.isForward == isForward;
 		}
 		return false;
 	}
 
-	private List findLeavesBetweenInclusive(EditPart left, EditPart right) {
+	private static List<EditPart> findLeavesBetweenInclusive(EditPart left, EditPart right) {
 		if (left == right)
 			return Collections.singletonList(left);
 		EditPart commonAncestor = ToolUtilities.findCommonAncestor(left, right);
 
 		EditPart nextLeft = left.getParent();
-		List children;
+		List<? extends EditPart> children;
 
-		ArrayList result = new ArrayList();
+		List<EditPart> result = new ArrayList<>();
 
 		if (nextLeft == commonAncestor)
 			result.add(left);
 		while (nextLeft != commonAncestor) {
 			children = nextLeft.getChildren();
 			for (int i = children.indexOf(left); i < children.size(); i++)
-				depthFirstTraversal((EditPart) children.get(i), result);
+				depthFirstTraversal(children.get(i), result);
 
 			left = nextLeft;
 			nextLeft = nextLeft.getParent();
 		}
 
-		ArrayList rightSide = new ArrayList();
+		List<EditPart> rightSide = new ArrayList<>();
 		EditPart nextRight = right.getParent();
 		if (nextRight == commonAncestor)
 			rightSide.add(right);
 		while (nextRight != commonAncestor) {
 			children = nextRight.getChildren();
-			int end = children.indexOf(right);
-			for (int i = 0; i <= end; i++)
-				depthFirstTraversal((EditPart) children.get(i), rightSide);
+			for (int i = 0; i <= children.indexOf(right); i++)
+				depthFirstTraversal(children.get(i), rightSide);
 
 			right = nextRight;
 			nextRight = nextRight.getParent();
 		}
 
 		children = commonAncestor.getChildren();
 		int start = children.indexOf(left) + 1;
-		int end = children.indexOf(right);
-		for (int i = start; i < end; i++)
-			depthFirstTraversal((EditPart) children.get(i), result);
+		for (int i = start; i < children.indexOf(right); i++)
+			depthFirstTraversal(children.get(i), result);
 
 		result.addAll(rightSide);
 		return result;
 	}
 
-	private List findNodesBetweenInclusive(EditPart left, EditPart right) {
+	private static List<EditPart> findNodesBetweenInclusive(EditPart left, EditPart right) {
```

Comment:
Are you sure transferring this method to a static context will not break anything?

Reply:
As it is private and not accessing any fields and it was suggested by the Eclipse Java compiler I'm pretty sure.

Reply:
As this method is private, not accessing any fields, and as it was suggested by the Eclipse Java compiler I'm pretty sure. But if you are not feeling comfortable I'm happy to change back.